### PR TITLE
Changed console.lua to fix vulnerability

### PR DESCRIPTION
**Pull request by JanTuck**
So a while back i talked to Divran about this vulnerability. 

Basically, when you run this string inside an e2 `concmd("say " + "Hej":repeat(5000000000))` it crashes most servers.

### CODE REVIEW

**Comment by Divran** [comment]
would it not make more sense to put this inside `validConCmd`, or does it not crash if used in the other functions?

**Comment by adosikas** [comment]
Shouldn't this cause a tick qouta anyway? Repeat should use 0.01 ops per output char.

**Comment by JanTuck** [comment]
@adosikas Yeah, i thought it would too but on a very restrictive server, it ran the command and crashed anyway.

@Divran prolly true.

**Comment by bigdogmat** [comment]
> Shouldn't this cause a tick qouta anyway? Repeat should use 0.01 ops per output char.

Manual changes to `prf` are only checked at the start of `seq` statements and every `0.030303` seconds, this would execute before it hits quota.

**Comment by adosikas** [comment]
In that case would it not make sense to have a dedicated addPerf( ... ) function that does the check as well (*before* the action is done ofc)?

**Comment by bigdogmat** [comment]
Yeah, I talked about this [here](https://github.com/wiremod/wire/pull/1497#issuecomment-350941980).

**Comment by JanTuck** [comment]
Well, even if there was added a addPerf before, you could still just copy a 5million long string into the e2? If you are dedicated that is. Haha

**Comment by Nebual** [comment]
Prf discussion is interesting, feel free to continue it in #1497 or wherever, though I'll merge this PR since why not

File: lua/entities/gmod_wire_expression2/core/console.lua [hunk 1]
```diff
@@ -8,10 +8,12 @@ local function validConCmd(self, command)
...
 	if not ply:IsValid() then return false end
 	if ply:GetInfoNum("wire_expression2_concmd", 0) == 0 then return false end
-
+  	-- Validating the concmd length to ensure that it won't crash the server.
+  	if #command > 500000 then return false end
+	
 	local whitelist = (ply:GetInfo("wire_expression2_concmd_whitelist") or ""):Trim()
 	if whitelist == "" then return true end
-
+
```
Comment (oldmud0) [inline lua/entities/gmod_wire_expression2/core/console.lua]: 🙁

Comment (CaptainPRICE) [reply lua/entities/gmod_wire_expression2/core/console.lua]: :(

File: lua/entities/gmod_wire_expression2/core/console.lua [hunk 2]
```diff
@@ -8,10 +8,12 @@ local function validConCmd(self, command)
 	local ply = self.player
 	if not ply:IsValid() then return false end
 	if ply:GetInfoNum("wire_expression2_concmd", 0) == 0 then return false end
-
+  	-- Validating the concmd length to ensure that it won't crash the server.
+  	if #command > 500000 then return false end
```
Comment (CaptainPRICE) [inline lua/entities/gmod_wire_expression2/core/console.lua]: This PR has whitespace issues... Why merge so quick @Nebual

Comment (JanTuck) [reply lua/entities/gmod_wire_expression2/core/console.lua]: My bad 😅

**Pull request merged**